core/dnsserver/onstartup_test.go

```diff
@@ -0,0 +1,14 @@
+package dnsserver
+
+import "testing"
+
+func TestCheckDomainName(t *testing.T) {
+
+	// Note: checkDomainName() returns true for RFC1035 preferred syntax.
+	if !checkDomainName("an-example.com.") {
+		t.Fatalf(`Expected true but received false for "an-example.com"`)
+	}
+	if checkDomainName("example-?&^%$.com") {
+		t.Fatalf(`Expected false but received true for "example-?&^$.com"`)
+	}
+}
```

Comment:
You need more test cases here.  At the very least you should add the root test case, since your regex didn't account for it initially.